doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -373,7 +375,16 @@ the propositional implication and function types.
 Applications
 ------------
 
-:n:`@term__fun @term` denotes applying the function :n:`@term__fun` to :token:`term`.
+:n:`@term__fun @term` denotes applying the :term:`function`
+:n:`@term__fun` to :token:`term`.
+
+.. glossary::
+
+   Function
+     Any Coq object whose type is a :ref:`product <product>`.
+
+     In particular, a non-dependent function has a non-dependent
+     product type :n:`@type -> @type`.
 
 :n:`@term__fun {+ @term__i }` denotes applying
```

Comment:
OTOH, indexes are old-fashioned concept.  Instead of 
- clicking on "Indexes",
- clicking on, say, "Tactic index",
- clicking on a letter,
- scanning the list and
- clicking on the tactic you're interested in,

we should create a popup search widget.  I should be able to press CTRL-T, then type the first few characters of the tactic name with a dropdown list showing the matches from which I can jump directly to what I want.  Of course the dropdown updates as I type each character.  CTRL-C for commands, etc.  A widget would also keep the browser history cleaner; to go back after a lookup is just the "Back" button once while with the current indexes it's 2-3 times.

This should be fairly routine JavaScript.  I expect we'd have to duplicate the index info so that it's accessible to JavaScript (AFAIK JavaScript isn't able to read the HTML source for a page).  And there's probably a way to cache that data so you don't have to download it every time you go from one file to another.

And while I'm on a roll, it would also be nice if the browser history could be customized so it told you that you were looking at the "Function" command on the previous page (based on jumping to that link) instead of just the generic "The Gallina specification language ...".  I'm not sure that's possible.

![image](https://user-images.githubusercontent.com/1253341/79611814-45b04200-80b0-11ea-8671-fcd36c65614b.png)

Reply:
That'd be nice. I'd rather not create a specific shortcut for each kind of searchable. Rather, I would override the current search bar to automatically show results as you type like you said. When typing something it would show you one tactic, one command, one flag, etc. that match with some light but visible description of the kind of object so that you can quickly jump to the one you're interested in.